src/sentry/integrations/jira_server/client.py

```diff
@@ -1,19 +1,223 @@
-from urllib.parse import parse_qsl
+import logging
+import re
+from urllib.parse import parse_qsl, urlparse
 
 from django.urls import reverse
 from oauthlib.oauth1 import SIGNATURE_RSA
 from requests_oauthlib import OAuth1
 
 from sentry.integrations.client import ApiClient
-from sentry.integrations.jira.client import JiraApiClient
 from sentry.shared_integrations.exceptions import ApiError
 from sentry.utils import jwt
 from sentry.utils.http import absolute_uri
 
+logger = logging.getLogger("sentry.integrations.jira")
+
+JIRA_KEY = f"{urlparse(absolute_uri()).hostname}.jira"
+ISSUE_KEY_RE = re.compile(r"^[A-Za-z][A-Za-z0-9]*-\d+$")
+CUSTOMFIELD_PREFIX = "customfield_"
+
+
+class JiraServerClient(ApiClient):
```

Comment:
Probably more of these will be deprecated or updated in the upcoming PR to fix Jira Server for newer versions, but for now I just copied over what was in the Cloud one.

Reply:
Since there's gonna be a follow up PR, I'll won't add any comments about JiraCloud specific code here as well 🚀